File: MvvmAndroid/src/main/java/com/ibot/cyranosystems/mvvmandroid/MainActivity.java
package com.ibot.cyranosystems.mvvmandroid;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import com.ibot.cyranosystems.mvvmandroid.commonds.UserLogin;
import com.ibot.cyranosystems.mvvmandroid.databinding.ActivityMainBinding;
import com.ibot.cyranosystems.mvvmandroid.model.User;
import com.ibot.cyranosystems.mvvmandroid.viewmodel.UserModel;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding activityMainBinding;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
       // setContentView(R.layout.activity_main);
        activityMainBinding=DataBindingUtil.setContentView(this,R.layout.activity_main);
      setHintEdittext();
      callLoginEvent();
    }

    private void callLoginEvent() {
        activityMainBinding.setUserloginevent(new UserLogin() {
            @Override
            public void onClickLogin() {
                Toast.makeText(MainActivity.this,activityMainBinding.getLogin().getEmail(),Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void setHintEdittext() {
        /* Databinding set Hint in Edittext*/
        UserModel userModel=new UserModel(new User("email","password"));
        activityMainBinding.setLogin(userModel);
    }
}

File: KotlinFirstApp/src/main/java/com/ibot/cyranosystems/kotlinfirstapp/MainActivity.kt
package com.ibot.cyranosystems.kotlinfirstapp

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        loginClick.setOnClickListener{
            var usernameval:String=edtUsername.text.toString()
            var userpassval:String=edtPassword.text.toString()
            if(usernameval.equals("")){
                showToast("Please enter username")
            }else if(userpassval.equals("")){
                showToast("Please enter password")
            }else{
                showToast("Username is "+usernameval+ " user Password is "+userpassval)
                val intent = Intent(this, SimpleListViewActivity::class.java)
                startActivity(intent)
            }
        }
    }
    fun showToast(args:String){
        Toast.makeText(this,args,Toast.LENGTH_SHORT).show()
    }
}
